hadoop-gremlin/src/main/java/org/apache/tinkerpop/gremlin/hadoop/structure/io/graphson/GraphSONRecordReader.java

```diff
@@ -56,11 +59,33 @@ public GraphSONRecordReader() {
     public void initialize(final InputSplit genericSplit, final TaskAttemptContext context) throws IOException {
         this.lineRecordReader.initialize(genericSplit, context);
         this.hasEdges = context.getConfiguration().getBoolean(Constants.GREMLIN_HADOOP_GRAPH_READER_HAS_EDGES, true);
-        this.graphsonReader = GraphSONReader.build().mapper(
-                GraphSONMapper.build().
-                        version(GraphSONVersion.valueOf(context.getConfiguration().get(Constants.GREMLIN_HADOOP_GRAPHSON_VERSION, "V3_0"))).
-                        typeInfo(TypeInfo.PARTIAL_TYPES).
-                        addRegistries(IoRegistryHelper.createRegistries(ConfUtil.makeApacheConfiguration(context.getConfiguration()))).create()).create();
+        GraphSONVersion graphSONVersion =
+                GraphSONVersion.valueOf(context.getConfiguration().get(Constants.GREMLIN_HADOOP_GRAPHSON_VERSION, "V3_0"));
+        Mapper mapper = null;
+
+        if (graphSONVersion == GraphSONVersion.V2_0) {
```

Comment:
might be nice if you make this code a bit more compact - like, create a builder instance with the common things and then just `addCustomModule(...)` in the if-then parts. i can do it after it's merged if you don't have time.

Reply:
I can do that.

Reply:
DONE.